DOCS-7855-7856-4.3-Changes-for-SDK (#63)
mulesoft/docs-mule-sdk: modules/ROOT/pages/reconnection.adoc

<2> Communicating to the `SourceCallback` the `ConnectionException` containing
the failure connection that should be replaced.

> ==== Accessing the Retry Policy Template from within an Operation

Reviewer: ```suggestion
==== Accessing the Retry Policy Template from Within an Operation
```

Available since version 1.3.

> There are advanced cases in which an operation may need to access the RetryPolicyTemplate object that has been configured for it. For example, an operation might be leveraging an existing library which handles retries on its own, or your use case might require applying retries without necessarily reconnecting.

Reviewer: ```suggestion
There are advanced cases in which an operation might need to access the `RetryPolicyTemplate` object that has been configured for it. For example, an operation might be leveraging an existing library that handles retries independently, or your use case might require applying retries without necessarily reconnecting.
```

> To support these cases, operation can request the effective RetryPolicyTemplate instance being used by adding it as an operation parameter. Consider this example from the WebSockets connector:

Reviewer: ```suggestion
To support these cases, operations can request that the effective `RetryPolicyTemplate` instance be used by adding it as an operation parameter. Consider this example from Anypoint Connector for WebSockets (WebSockets Connector):
```

[source, java, linenums]
----
public void send(String socketId,
                   @Content TypedValue<InputStream> content,
                   @Connection WebSocketConnection conn,
                   CompletionCallback<Void, Void> callback,
                   RetryPolicyTemplate retryPolicyTemplate) {

                   }
----

> The retryPolicyTemplate argument is not translated into an operation parameter, but it is injected with the effective value upon execution.

Reviewer: "injected"?

[NOTE]

> This feature is intended for advanced scenarios only. Typically, you should  throw a ConnectionException and allow the Mule Runtime handle reconnection and retries automatically. Only use this feature when absolutely necessary.

Reviewer: ```suggestion
This feature is intended for advanced scenarios only. Typically, you should throw a ConnectionException and allow Mule to handle reconnection and retries automatically. Use this feature only when absolutely necessary.
```
